bpo-43234: Prohibit non-ThreadPoolExecutor in loop.set_default_executor (#24540)
python/cpython: Doc/library/asyncio-eventloop.rst

.. method:: loop.set_default_executor(executor)

   Set *executor* as the default executor used by :meth:`run_in_executor`.
   *executor* must be an instance of
   :class:`~concurrent.futures.ThreadPoolExecutor`.

>    .. deprecated-removed:: 3.8 3.10

Reviewer: ```suggestion
   .. versionchanged:: 3.11
```

>       Accepting an ``executor`` that is not an instance of
>       :class:`~concurrent.futures.ThreadPoolExecutor`.

Reviewer: ```suggestion
      *executor* must be an instance of
      :class:`~concurrent.futures.ThreadPoolExecutor`.
```
